runelite-client/src/main/java/net/runelite/client/plugins/camera/CameraPlugin.java

```diff
@@ -117,36 +128,84 @@ protected void shutDown()
 		controlDown = false;
 	}
 
+	@Subscribe
+	public void onGameTick(GameTick gameTick)
+	{
+		final int timeoutTicks = (config.compassMultipleClickTimeout() * 1000) / Constants.GAME_TICK_LENGTH;
+		if (currentLeftClickOption.equals(defaultLeftClickOption) || (client.getTickCount() < (lastUpdatedTick + timeoutTicks)))
+		{
+			return;
+		}
+		currentLeftClickOption = defaultLeftClickOption;
+	}
+
 	@Subscribe
 	public void onMenuEntryAdded(MenuEntryAdded menuEntryAdded)
 	{
-		if (menuEntryAdded.getType() == MenuAction.CC_OP.getId() && menuEntryAdded.getOption().equals(LOOK_NORTH) && config.compassLook())
+		if (menuEntryAdded.getType() == MenuAction.CC_OP.getId())
 		{
-			MenuEntry[] menuEntries = client.getMenuEntries();
-			int len = menuEntries.length;
-			MenuEntry north = menuEntries[len - 1];
+			if (ArrayUtils.indexOf(options, menuEntryAdded.getOption()) < 0)
+			{
+				return;
+			}
 
-			menuEntries = Arrays.copyOf(menuEntries, len + 3);
+			final int count = options.length;
+			MenuEntry[] menuEntries = client.getMenuEntries();
+			int currentOptionIndex = ArrayUtils.indexOf(options, currentLeftClickOption);
 
-			// The handling for these entries is done in ToplevelCompassOp.rs2asm
-			menuEntries[--len] = createCameraLookEntry(menuEntryAdded, 4, LOOK_WEST);
-			menuEntries[++len] = createCameraLookEntry(menuEntryAdded, 3, LOOK_EAST);
-			menuEntries[++len] = createCameraLookEntry(menuEntryAdded, 2, LOOK_SOUTH);
-			menuEntries[++len] = north;
+			if (config.compassLook())
+			{
+				menuEntries = Arrays.copyOf(menuEntries, menuEntries.length + 3);
```

Comment:
`3` should instead be a variable or reference to a variable, such as `OPTIONS.length - 1` since it makes no sense to be a magic number.